src/main/java/com/blog/web/controllers/AuthController.java

```diff
@@ -0,0 +1,64 @@
+package com.blog.web.controllers;
+
+import com.blog.web.dto.RegistrationDto;
+import com.blog.web.models.UserEntity;
+import com.blog.web.services.UserService;
+import jakarta.validation.Valid;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+
+@Controller
+public class AuthController {
+    private UserService userService;
+
+    public AuthController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping("/login")
+    public String loginPage() {
+        return "auth/login";
+    }
+
+    @GetMapping("/register")
+    public String getRegisterForm(Model model) {
+        RegistrationDto user = new RegistrationDto();
+        model.addAttribute("user", user);
+        return "auth/register";
+    }
+
+    @PostMapping("/register/save")
+    public String register(@Valid @ModelAttribute("user")RegistrationDto user,
+                           BindingResult result,
+                           Model model) {
+        UserEntity existingUserEmail = userService.findByEmail(user.getEmail());
+        if(
+                existingUserEmail != null &&
```

Comment:
repitition on the if check might be worth making another function, also take a look at StringUtils.isblank() might be what your looking for